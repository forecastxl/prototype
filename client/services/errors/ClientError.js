function ClientError(message) {
  this.name = 'ClientError'
  this.message = message || 'Something went wrong but no message was provided'
}

ClientError.prototype = Error.prototype

export default ClientError
