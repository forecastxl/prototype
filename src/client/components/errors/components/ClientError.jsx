import React, { PropTypes } from 'react'

function ClientError({ errorMessage }) {
  return (
    <div>
      Something went wrong with this application. The errormessage was: {errorMessage}.
    </div>
  )
}

ClientError.propTypes = {
  errorMessage: PropTypes.string.isRequired
}

export default ClientError
