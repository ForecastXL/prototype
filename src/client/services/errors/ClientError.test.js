import ClientError from './ClientError'

describe('ClientError', () => {
  it('should return a valid error', () => {
    const actual = new ClientError()

    expect(actual).toBeInstanceOf(Error)
  })

  it('should accept a message', () => {
    const message = 'message'
    const actual = new ClientError(message)

    expect(actual.message).toBe(message)
  })

  it('should fall back to a default message', () => {
    const actual = new ClientError()

    expect(actual.message).toMatchSnapshot()
  })
})
