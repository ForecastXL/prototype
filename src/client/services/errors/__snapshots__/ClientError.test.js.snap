// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientError should fall back to a default message 1`] = `"Something went wrong but no message was provided"`;
